test/process_stealing_test.ml

```diff
@@ -0,0 +1,33 @@
+[@@@warning "-8"]
+
+open Riot
+
+type Message.t += A
+
+let rec loop pid msg n =
+  if n = 0 then send pid msg
+  else (
+    yield ();
+    loop pid msg (n - 1))
+
+let main () =
+  Riot.run @@ fun () ->
+  let this = self () in
+
+  let _ =
+    spawn_pinned (fun () ->
```

Comment:
We could try having the first process do something more aggressive, like a `while true do .. done` that only yields every 1000 iterations or so.